From 44f781880acb14209b77d33ce7e83df318a932a6 Mon Sep 17 00:00:00 2001
From: Mei Xie <quic_meixie@quicinc.com>
Date: Thu, 31 Jul 2025 14:17:55 +0800
Subject: [PATCH 15/28] PENDING: net: stmmac: Change PDMA mapping to resolve
 SMMU crash

Change mapping from PDMA7 <-> TC5 to PDMA7 <-> TC6 to match SMMU
to avoid SMMU crash.

Change-Id: Ic7a0f9db4ba4d71dd0e772352776ae6392ad1015
Signed-off-by: Mei Xie <quic_meixie@quicinc.com>
---
 drivers/net/ethernet/stmicro/stmmac/dwmac-qcom-ethqos.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/dwmac-qcom-ethqos.c b/drivers/net/ethernet/stmicro/stmmac/dwmac-qcom-ethqos.c
index 0a6a007f40ddc..bef79c680ef60 100644
--- a/drivers/net/ethernet/stmicro/stmmac/dwmac-qcom-ethqos.c
+++ b/drivers/net/ethernet/stmicro/stmmac/dwmac-qcom-ethqos.c
@@ -844,7 +844,7 @@ static void qcom_ethqos_hdma_cfg(struct plat_stmmacenet_data *plat)
 	plat->dma_cfg->tx_pdma_map[4] = 4;
 	plat->dma_cfg->tx_pdma_map[5] = 5;
 	plat->dma_cfg->tx_pdma_map[6] = 5;
-	plat->dma_cfg->tx_pdma_map[7] = 5;
+	plat->dma_cfg->tx_pdma_map[7] = 6;
 	plat->dma_cfg->tx_pdma_map[8] = 6;
 	plat->dma_cfg->tx_pdma_map[9] = 6;
 	plat->dma_cfg->tx_pdma_map[10] = 7;
@@ -858,7 +858,7 @@ static void qcom_ethqos_hdma_cfg(struct plat_stmmacenet_data *plat)
 	plat->dma_cfg->rx_pdma_map[4] = 4;
 	plat->dma_cfg->rx_pdma_map[5] = 5;
 	plat->dma_cfg->rx_pdma_map[6] = 5;
-	plat->dma_cfg->rx_pdma_map[7] = 5;
+	plat->dma_cfg->rx_pdma_map[7] = 6;
 	plat->dma_cfg->rx_pdma_map[8] = 6;
 	plat->dma_cfg->rx_pdma_map[9] = 6;
 	plat->dma_cfg->rx_pdma_map[10] = 7;
